### PR TITLE
Issue #49 circular linked list

### PR DESCRIPTION
**Pull request by prshnt19**
Added SinglyCircularLinkedList and DoublyCircularLinkedList
#### References to other Issues or PRs or Relevant literature

#49

### CODE REVIEW

**Comment by czgdp1807** [comment]
Please resolve the conflicts.

**Comment by prshnt19** [comment]
Resolved.

File: pydatastructs/linear_data_structures/linked_lists.py [hunk 1]
```diff
@@ -508,3 +512,219 @@ def __getitem__(self, index):
...
+    >>> scll.extract(1).data
+    6
+    >>> print(scll)
+    [7.2, 5]
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Linked_list#Circular_linked_list
+    .. [2] https://www.geeksforgeeks.org/circular-singly-linked-list-insertion/
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/linked_lists.py]:
```suggestion
```

Comment (prshnt19) [reply pydatastructs/linear_data_structures/linked_lists.py]: The wiki page does not have enough info so I added other references.

**Comment by czgdp1807** [comment]
@prshnt19 If you want, you can add your name to AUTHORS file in a separate PR.

File: pydatastructs/linear_data_structures/linked_lists.py (hunk 1 above)
Comment (prshnt19) [reply pydatastructs/linear_data_structures/linked_lists.py]: The info on the Wiki is not enough. Should I add a reference from other sites or leave it as it is?

Comment (czgdp1807) [reply pydatastructs/linear_data_structures/linked_lists.py]: You can add links to university lectures.

Comment (prshnt19) [reply pydatastructs/linear_data_structures/linked_lists.py]: Update Wiki References.

File: pydatastructs/linear_data_structures/linked_lists.py [hunk 2]
```diff
@@ -508,3 +512,217 @@ def __getitem__(self, index):
...
         return current_node
+
+class SinglyCircularLinkedList(SinglyLinkedList):
+    """
+    Represents Singly Circular Linked List.
+
+    Parent Class
+    ============
+
+    SinglyLinkedList.
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/linked_lists.py]:
```suggestion
```

**Comment by czgdp1807** [comment]
LGTM. 
@codezonediitj/gssoc2020 Please see if the tests are correct.

**Comment by prshnt19** [comment]
Tests are correct. I have verified it.

**Pull request merged**